docs/edgeql/operators/type.rst

```diff
@@ -35,8 +35,54 @@ IS
         ... FILTER User.name = 'Alice';
         {True}
 
-        db> SELECT User IS (Text, Named);
-        {True, ..., True}  # one for every user instance
+.. eql:operator:: TYPEUNION: A | B
+
+    :optype A: object type
+    :optype B: object type
+    :resulttype: union type
+    :index: union type
+
+    Create a *union type* from types ``A`` and ``B``.
+
+    A *union type* is a way to describe the type of a set resulting from a
```

Comment:
It's the other way around. The type of the result of a `UNION` expression is a union type, but we don't define union types through a `UNION`.  It is also not exclusive to the object types. There's a better definition in #219.

Reply:
I'll change it to say that the type of the result of a `UNION` of two objects is an *example* of a union type.

Although #219 has indeed a more complete description that's also the problem. Currently, we don't have most of that implemented, so in the spirit of describing what actually is, I described the specific usages that are implemented and valid: link targets, `UNION` results and `IS`. These features were present even before the syntax change and generalization of type operations.

Type intersection is not implemented at all. Neither are type unions of scalar types.

Reply:
Unions scalar types _are_ implemented, otherwise it wouldn't be possible to do a `UNION` over scalars.  A union of two scalar types is their common implicitly-castable-to type.